### DISCUSSION TITLE
`mlflow.autolog()` not working as expected

### DISCUSSION THREAD

**Discussion by retzero** [mlflow/mlflow#16604] [Q&A] [answered]
Hello.

I'm trying to log metrics with this simple pytorch model: 
- https://github.com/pytorch/examples/tree/main/mnist

I've added `mlflow.autolog()` to main.py as below but no parameters, no metrics are recorded to the local MLflow server.

  ```diff
    from torch.optim.lr_scheduler import StepLR
  
  + import mlflow
  + mlflow.autolog()
  
    class Net(nn.Module):
        def __init__(self):
  ```

To loss and accuracy be logged, I have to add it into training code explicitely;
```diff
+            mlflow.log_metrics({
+                'train_loss': loss.item(),
+                'train_accuracy': accuracy
+            }, step=epoch * len(train_loader) + batch_idx)
```

Why `autolog()` fail to log parameters and metrics automatically?

Thanks.

**Comment by AndhikaWB** [comment, marked as answer]
MLFlow doesn't have an autolog support for vanilla PyTorch, you are free to design your own training loop to begin with. MLFlow can't magically know which one is a metric and which ones are not.

TensorFlow and scikit-learn have a standard method to call for training the model, that's why autolog will work for those libraries.

You either have to call `log_metrics` and `log_model` manually, or use PyTorch Lightning (this is the one supported by MLFlow).

If you are a beginner want to learn in-depth ML, I recommend familiarizing yourself with vanilla PyTorch first. However, if you don't care about that and just want to quickly build something, you can try PyTorch Lightning directly.

Note that despite the similarity in name, PyTorch Lightning is not affiliated with PyTorch or Facebook.
